ComSemApp/templates/ComSemApp/student/assessment_results.html

```diff
@@ -80,10 +80,97 @@ <h1 class="card-title">Review complete!</h1>
+                    transcription error before requesting review.<br>
+                    If you have already done so, continue. 
+                </span>
+            </div>
+            <div class="modal-footer">
+                <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
+                <button type="button" class="btn btn-warning" id="confirm-button">Request Review</button>
+            </div>
+        </div>
+    </div>
+</div>
+  
+  
 <div class="btn-group d-flex card-footer" role="group" aria-label="...">
     <a href="{% url 'student:course' course.id %}" class ="btn-success btn w-100 m-1">Return Home</a>
     <a href="{% url 'student:course' course.id %}" class ="btn-success btn w-100 m-1">Try Again</a>
 </div>
+
+<!-- page form holder, input specifies url "speaking_practice_teacher_request" which links to SpeakingPracticeTeacherReviewRequest" -->
+<form id="pageForm" method="POST">
+    <input hidden type="text" id="teacher_review_request" value="{% url 'student:speaking_practice_teacher_request' course.id%}">
+  </form>
+
 <script src="{% static "ComSemApp/js/ComSemRecording-opus.js" %}?{% now "U" %}"></script>
+<script>
+
+// Add click event listener to all request buttons
+$(".request-btn").on("click", function() {
+    // Get the request ID from the data attribute of the button
+    var attempt_id = $(this).data("submit-id");
+  
+    // Show the modal dialog box
+    $("#confirm-modal").modal("show");
+  
+    // Add click event listener to the confirm button
+    $("#confirm-button").one("click", function() {
+      // Submit the form with the submit ID as a parameter
+      console.log("confirm clicked")
+        teacherRequestClick(attempt_id);
+      // Hide the modal dialog box
+      $("#confirm-modal").modal("hide");
+    });
+  });
+
+function teacherRequestClick(attempt){
+    if (!$("#request" + attempt).hasClass("disabled")){
+        console.log(attempt);
+        $("#request" + attempt).removeClass("btn-warning").addClass("btn-secondary").prop("disabled", true);
+
+        let requestFormData = new FormData(); // creates new form
+        requestFormData.append( 'attempt_id', attempt); 
+        //requestFormData.append('student', student);
+        var csrf_token = '{{ csrf_token }}'; // grabs csrf_token for certification
+        
+
+        $.ajax({ // send POST request
+            type: "POST",
+            url: $("#teacher_review_request").val(), // links to html form above which specifies the url to recieve request
+            data: requestFormData,
+            processData: false,
+            contentType: false,
+            timeout: 5000, // if takes longer than 5 secs, return timeout error
+            headers: { // places token in request so that it is certified by server
+                'X-CSRFToken': csrf_token
+            },
+            success:(response) => {
+                cs_notification('success', "Request Sent")
+                resolve();
```

Comment:
These `resolve()` and `reject()` calls should either be removed or this function needs to return a promise.